MonitorControl/Support/Utils.swift

```diff
@@ -50,33 +50,36 @@ class Utils: NSObject {
     display.saveMaxValue(Int(maxValue), for: command)
 
     os_log("%{public}@ (%{public}@):", type: .info, display.name, String(reflecting: command))
-    os_log(" - current ddc value: %{public}@", type: .info, String(currentDDCValue))
-    os_log(" - maximum ddc value: %{public}@", type: .info, String(maxValue))
+    os_log(" - current ddc value: %{public}@ - from display? %{public}@", type: .info, String(currentDDCValue), String(values != nil))
+    os_log(" - maximum ddc value: %{public}@ - from display? %{public}@", type: .info, String(maxValue), String(values != nil))
 
     if command != .audioSpeakerVolume {
       slider.integerValue = Int(currentDDCValue)
       slider.maxValue = Double(maxValue)
     } else {
       // If we're looking at the audio speaker volume, also retrieve the values for the mute command
       var muteValues: (current: UInt16, max: UInt16)?
+
       os_log("Polling %{public}@ times", type: .info, String(tries))
+      os_log("%{public}@ (%{public}@):", type: .info, display.name, String(reflecting: DDC.Command.audioMuteScreenBlank))
 
       if tries != 0 {
         muteValues = display.readDDCValues(for: .audioMuteScreenBlank, tries: tries, minReplyDelay: delay)
       }
 
-      os_log("%{public}@ (%{public}@):", type: .info, display.name, String(reflecting: DDC.Command.audioMuteScreenBlank))
+      if let muteValues = muteValues {
+        os_log(" - current ddc value: %{public}@", type: .info, String(muteValues.current))
+        os_log(" - maximum ddc value: %{public}@", type: .info, String(muteValues.max))
 
-      if muteValues != nil {
-        os_log(" - current ddc value: %{public}@", type: .info, String(muteValues!.current))
-        os_log(" - maximum ddc value: %{public}@", type: .info, String(muteValues!.max))
-
-        display.saveValue(Int(muteValues!.current), for: .audioMuteScreenBlank)
-        display.saveMaxValue(Int(muteValues!.max), for: .audioMuteScreenBlank)
+        display.saveValue(Int(muteValues.current), for: .audioMuteScreenBlank)
+        display.saveMaxValue(Int(muteValues.max), for: .audioMuteScreenBlank)
+      } else {
+        os_log(" - current ddc value: unknown", type: .info)
+        os_log(" - stored maximum ddc value: %{public}@", type: .info, String(display.getMaxValue(for: .audioMuteScreenBlank)))
       }
 
       // If the system is not currently muted, or doesn't support the mute command, display the current volume as the slider value
-      if muteValues == nil || muteValues!.current == 2 {
```

Comment:
just letting you know that 
```swift
if muteValues == nil || muteValues!.current == 2 {
```
is not the same as 
```swift
if muteValues == nil, muteValues?.current == 2 {
```
When using `,` Swift only executes the body if **all** the assignments are properly completed (so it translates more like `&&`). The reason for this comma afaik is to be able to indicate that you can use the previously unwrapped value in your next unwrap like this:
```swift
if let oppositeKey = oppositeKey, let oppositeKeyTimer = self.keyRepeatTimers[oppositeKey], oppositeKeyTimer.isValid {														    ^
```
so when you use comma, you can use the `oppositeKey` you just unwrapped in your next unwrap.

Not sure if that was an intended change or not, just letting you know in case it wasn't ;)

Reply:
Whoops! Nope, wasn't intended.

Changed it back to:
```swift
if muteValues == nil || muteValues!.current == 2
```

Force-unwrapping seems alright here since we're specifically checking whether **muteValues** is nil as a condition of the if statement.